client/src/pages/NotFound/index.tsx

```diff
@@ -0,0 +1,45 @@
+import { Link, useNavigate } from "react-router-dom";
+import { Error, Home, ArrowBack } from "@mui/icons-material";
+import Header from "../../components/Header";
+
+function NotFound() {
+  const navigate = useNavigate();
+
+  return (
+    <>
+      <Header />
+      <main className="flex justify-center">
+        <div className="flex min-h-[calc(100vh-6.875rem)] w-full max-w-7xl flex-col items-center justify-center gap-4 p-4 md:min-h-[calc(100vh-4.5rem)]">
+          <h1 className="text-center text-xl font-bold italic text-gray-700">
+            Sorry, the page requested does not exist
+          </h1>
+          <div className="flex gap-4">
+            <div
+              className="flex cursor-pointer items-center justify-center gap-2 rounded-lg border border-sky-300 bg-sc-blue px-4 py-2 text-white hover:bg-sc-dark-blue"
+              onClick={() => navigate(-1)}
```

Comment:
this is a good way to go back in history but consider adding a check if there's any history available, otherwise redirect to the home page or another fallback route.

Reply:
That makes a lot of sense! Looking into how to implement this feature.